.sandbox/auto_scale.py

```diff
@@ -0,0 +1,350 @@
+    data,err = parse_json(result.stdout)
+    return data,err
+ 
+
+# Get buildfarm-workers containers in sandbox definition
+def workers_defined_in_template(template):
+    return list(filter(lambda container: container["name"].startswith("bf-worker"), template["containers"]))
+
+# Add one buildfarm-worker container to definition
+def alloc_worker_to_template(template):
+    workers = workers_defined_in_template(template)
+    for ii in range (0,max_worker):
+        worker_name = "bf-worker"+str(ii)
+        if not has_bf_worker(workers,worker_name):
+            worker = workers[0].copy()
+            worker["name"] = worker_name
+            template["containers"].append(worker)
+            return
+
+
+def has_bf_worker(workers,name):
+    for worker in workers:
+        if worker["name"] == name:
+            return True
+
+    return False
+
+
+# Scrape the metrics of buildfarm server.
+def scrape_buildfarm_server_metrics():
+    server_metrics = BuildfarmServerMetrics()
+    try:
+        response = requests.get(bf_server_metrics_url)
+        response.close()
+        if response.status_code != 200:
+            return None,response.text
+    except Exception as error:
+        return None,err
+    metrics = parse_metrics(response.text)
+    queue_size_key="queue_size"
+    worker_pool_size_key = "worker_pool_size"
+
+    if not queue_size_key in metrics:
+        return None, "metric {} not found".format(queue_size_key)
+    queue_size = 0
+    for k,v in metrics[queue_size_key].items():
+        server_metrics.queue_size += int(float(v))
+    if not worker_pool_size_key in metrics:
+        return None, "metric {} not found".format(worker_pool_size_key)
+
+    server_metrics.worker_pool_size = int(float(metrics[worker_pool_size_key]))
+    return server_metrics,None
+
+def parse_metrics(content):
+    buf = io.StringIO(content)
+    metrics = {}
+    while True:
+        line = buf.readline()
+        if len(line) == 0:
+            break
+        if line.startswith("#"):
+            continue
+        # Trim begin and end space
+        line = line.strip()
+        try:
+            index = line.rindex(" ")
+        except Exception as error:
+            continue
+        key = line[:index]
+        value = line[index+1:]
+        try:
+            label_index = key.index("{")
+        except Exception as error:
+            metrics[key] = value
+            continue
+        label = key[label_index:]
+        group=key[:label_index]
+        if group in metrics:
+            metrics[group][label] = value
+        else:
+            metrics[group]={label:value}
+    return metrics
+
+
+# Scrape the metrics of a buildfarm worker.
+def scrape_buildfarm_worker_metrics(worker_name):
+    url = "http://{}:9090/metric".format(worker_name)
+    worker_metrics = BuildfarmWorkerMetrics() 
+    worker_metrics.worker_name = worker_name
+
+    result = subprocess.run(["prom2json",url],capture_output=True, text=True)
```

Comment:
Where is `prom2json` installed? I saw it was removed from the script but I didn't see it's being added to setup.yaml.